### PR TITLE
feat(phonenumber): Add OCP wrapper for PhoneNumber utils library

### PR DESCRIPTION
**Pull request by nickvergessen**
* Required for https://github.com/nextcloud/spreed/issues/10410

## Summary

- [x] Extract current code that interacts with 3rdparty `libphonenumber` into an OCP wrapper, so it can be used savely by apps, e.g. Talk for the upcoming SIP dial-out.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required https://github.com/nextcloud/documentation/pull/11147
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
80cgu4

**Comment by nickvergessen** [comment]
Docs in https://github.com/nextcloud/documentation/pull/11147

File: lib/private/PhoneNumberUtil.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * @since 28.0.0
+ */
+class PhoneNumberUtil implements IPhoneNumberUtil {
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getCountryCodeForRegion(string $regionCode): ?int {
+		$phoneUtil = \libphonenumber\PhoneNumberUtil::getInstance();
+		$countryCode = $phoneUtil->getCountryCodeForRegion($regionCode);
+		return $countryCode === 0 ? null : $countryCode;
```
Comment (fsamapoor) [inline lib/private/PhoneNumberUtil.php:43]: Maybe I'm missing something here, but could it ever return `0` at all? I thought `null` is the return value that indicates an invalid/unsupported number.

Comment (nickvergessen) [reply lib/private/PhoneNumberUtil.php:43]: Upstream library returns int only and then 0 means "invalid, nothing matching found". I found this too hidden and opted for converting it to null in this case instead.

Comment (fsamapoor) [reply lib/private/PhoneNumberUtil.php:43]: Right. Thank you for the explanation Joas!

**Review by fsamapoor — APPROVED** [review]

**Pull request merged**